changelog.d/798.docker

```diff
@@ -0,0 +1 @@
+Update Dockerfile to specifically use Python 3.
```

Comment:
We don't have a `docker` entry in our pyproject.toml file which is why this is failing. I'm happy to either have this be a `misc`, or you can add a Docker section yourself.